### PR TITLE
add hdmf recipe

### PR DESCRIPTION
**Pull request by ajtritt**

### CODE REVIEW

**Comment by ajtritt** [comment]
@scopatz per #7715, can this just be merged?

**Comment by scopatz** [comment]
Since it is noarch: python, it doesn't need to be able to run on anything but circle

File: recipes/hdmf/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  noarch: python
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
```
Comment (scopatz) [inline recipes/hdmf/meta.yaml]: This should be as here: https://conda-forge.org/docs/meta.html#use-pip

File: recipes/hdmf/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    The Hierarchical Data Modeling Framework, or HDMF is a Python package
+    for working with hierarchical data. It provides APIs for specifying data
+    models, reading and writing data to different storage backends, and
+    representing data with Python object.
+  doc_url: http://hdmf.readthedocs.io/en/latest/
+
+extra:
+  recipe-maintainers:
+    - ajtritt
+    - oruebel
```
Comment (scopatz) [inline recipes/hdmf/meta.yaml]: @oruebel - are you OK with being a maintainer here?

File: recipes/hdmf/meta.yaml (hunk 1 above)
Comment (ajtritt) [reply recipes/hdmf/meta.yaml]: done

**Comment by scopatz** [comment]
OK just waiting on @oruebel and Circle

File: recipes/hdmf/meta.yaml (hunk 2 above)
Comment (oruebel) [reply recipes/hdmf/meta.yaml]: @scopatz yes, that is fine.

File: recipes/hdmf/meta.yaml (hunk 1 above)
Comment (ajtritt) [reply recipes/hdmf/meta.yaml]: For my own education, how is `PYTHON` supposed to be set? I assumed Jinja would pull it from the environment, but it wasn't able to find it.

File: recipes/hdmf/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
+{% set name = "hdmf" %}
+{% set version = "0.post0.dev1081" %}
+{% set sha256 = "8b0264a51ef1aea5fb22d450a7fa6cbc74bac97421e9667d1b7caee0ab74ce57" %}
+{% set PYTHON = "python" %}
```
Comment (scopatz) [inline recipes/hdmf/meta.yaml]: conda-build is supposed to set this to the full python path. You shouldn't need to do this.

Comment (ajtritt) [reply recipes/hdmf/meta.yaml]: Before I added that line, the Circle build failed for this reason: 

```
Error: Failed to render jinja template in /home/conda/conda-recipes/hdmf/meta.yaml:
'PYTHON' is undefined
Exited with code 1
```

https://circleci.com/gh/conda-forge/staged-recipes/34924?utm_campaign=workflow-failed&utm_medium=email&utm_source=notification

File: recipes/hdmf/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
+
+requirements:
+  build:
+    - python
+    - pip
+    - setuptools
+  host:
+    - pip
```
Comment (scopatz) [inline recipes/hdmf/meta.yaml]: I think you need python and setuptools here, and not in the build section

Comment (ajtritt) [reply recipes/hdmf/meta.yaml]: done

**Comment by ajtritt** [comment]
@scopatz is there anything else?

File: recipes/hdmf/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  sha256: {{ sha256 }}
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
+
+requirements:
+  build:
+    - pip
```
Comment (scopatz) [inline recipes/hdmf/meta.yaml]: You shouldn't need pip or the build section

Comment (ajtritt) [reply recipes/hdmf/meta.yaml]: removed!

**Pull request merged**